src/lib.rs

```diff
@@ -439,6 +450,98 @@ impl DstLayout {
         }
     }
 
+    /// Extends `self` with the layout of `field`, adding whatever padding is
```

Comment:
This comment should probably call out somewhere that the result of this method is only valid if the struct's field order is guaranteed. It might be worth writing a `# Safety` section documenting what correctness guarantees are provided, and with what preconditions.

Reply:
Following up on this, we may need to be very pedantic about the preconditions. It's roughly that the type needs to be `repr(C)`, but even that's not technically sufficient - e.g., a `repr(u8)` enum has a layout in which each of its variants is laid out _as if_ it was a `repr(C)` struct. Maybe instead of making reference to something like `repr(C)`, we can just *describe* the algorithm (ie, fields are laid out in order and separated by the minimum possible padding) and require that the type's layout follow that algorithm? Regardless of what approach we take, it will take some very careful and precise language to articulate when this method's result is correct.

Then, in the comments in the body, we may need to be more pedantic in order to articulate why the implementation is correct in terms of the pre- and post-conditions.